Update Pay  window style

aCorrado opened the pull request:
#### e7ccebfae181435e6a79e7c1cb29d9fd73c88652
<pre>
Update Pay  window style
<a href="https://bugs.webkit.org/show_bug.cgi?id=296933">https://bugs.webkit.org/show_bug.cgi?id=296933</a>
<a href="https://rdar.apple.com/147421439">rdar://147421439</a>

Reviewed by Aditya Keerthi.

Updated the sheet used by Apple Pay to be more cohesive with the new platform UI paradigm.

* Source/WebKit/Platform/spi/mac/AppKitSPI.h:
* Source/WebKit/Shared/ApplePay/mac/WebPaymentCoordinatorProxyMac.mm:
(WebKit::WebPaymentCoordinatorProxy::platformShowPaymentUI):

Canonical link: <a href="https://commits.webkit.org/298285@main">https://commits.webkit.org/298285@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/16d3edfe8ffc75e23d128abf706885bcf2d02d5b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/115021 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/34737 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/25218 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/121141 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/65670 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/35382 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/43298 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/87406 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/42217 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/117969 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/28198 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/103266 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/67802 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/27370 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/64793 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/97582 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/21503 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/124331 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/41994 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/31396 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/96203 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/42366 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/99456 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/95988 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/41187 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/19028 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/38005 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/18399 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/41869 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/47404 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/41413 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/44730 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/43149 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Source/WebKit/Platform/spi/mac/AppKitSPI.h:
```diff
@@ -77,6 +77,8 @@ typedef NS_OPTIONS(NSUInteger, NSWindowShadowOptions) {
     NSWindowShadowSecondaryWindow = 0x2,
 };
 
+static const NSWindowStyleMask NSWindowStyleMaskAlertWindow  = (NSWindowStyleMask)(1ull << 33ull);
```
```suggestion
static const NSWindowStyleMask NSWindowStyleMaskAlertWindow = (NSWindowStyleMask)(1ull << 33ull);
```

pxlcoder commented on Source/WebKit/Shared/ApplePay/mac/WebPaymentCoordinatorProxyMac.mm:
```diff
@@ -109,6 +110,7 @@
 
             ASSERT(!paymentCoordinatorProxy->m_sheetWindow);
             paymentCoordinatorProxy->m_sheetWindow = [NSWindow windowWithContentViewController:viewController];
+            paymentCoordinatorProxy->m_sheetWindow.get().styleMask = NSWindowStyleMaskAlertWindow | NSWindowStyleMaskTitled;
```
> to be more cohesive with the new platform UI paradigm

If this change is only intended for macOS 26, you'll need to introduce a conditional in `PlatformUse.h` and guard this logic, so it doesn't apply on older macOS.

aprotyas reviewed (commented):
A minor comment

pxlcoder commented on Source/WebKit/Shared/ApplePay/mac/WebPaymentCoordinatorProxyMac.mm:
```diff
@@ -109,7 +110,9 @@
             ASSERT(!paymentCoordinatorProxy->m_sheetWindow);
             paymentCoordinatorProxy->m_sheetWindow = [NSWindow windowWithContentViewController:viewController];
-
+#if HAVE(UI_GLASS_EFFECT)
```
```suggestion
#if HAVE(LIQUID_GLASS)
```

`HAVE(UI_GLASS_EFFECT)` is only true on iOS and tvOS (it refers to UIKit's `UIGlassEffect`).

pxlcoder reviewed (commented):
The commit message should be updated to simply be the following:

> Update Pay window style
> https://bugs.webkit.org/show_bug.cgi?id=296933
> rdar://147421439
> 
> Reviewed by NOBODY (OOPS!).
> 
> Updated the sheet used by Apple Pay to be more cohesive with the new platform UI paradigm.
> 
> * Source/WebKit/Platform/spi/mac/AppKitSPI.h:
> * Source/WebKit/Shared/ApplePay/mac/WebPaymentCoordinatorProxyMac.mm:
> (WebKit::WebPaymentCoordinatorProxy::platformShowPaymentUI):

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/86754d695aa713b966cb155c08c4865dae88c3c4. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48956

pxlcoder commented on Source/WebKit/Platform/spi/mac/AppKitSPI.h:
```diff
@@ -77,6 +77,10 @@ typedef NS_OPTIONS(NSUInteger, NSWindowShadowOptions) {
     NSWindowShadowSecondaryWindow = 0x2,
 };
 
+#if HAVE(UI_GLASS_EFFECT)
```
This exists independently of Liquid Glass, so I don't think it needs to be guarded?

Commit:
Update Pay  window style
https://bugs.webkit.org/show_bug.cgi?id=296933
rdar://147421439

Reviewed by Aditya Keerthi.

Updated the sheet used by Apple Pay to be more cohesive with the new platform UI paradigm.

* Source/WebKit/Platform/spi/mac/AppKitSPI.h:
* Source/WebKit/Shared/ApplePay/mac/WebPaymentCoordinatorProxyMac.mm:
(WebKit::WebPaymentCoordinatorProxy::platformShowPaymentUI):

Canonical link: https://commits.webkit.org/298285@main
